### PR TITLE
Separate unit tests into a different file when there are many of them

### PR DESCRIPTION
**Pull request by trent-reed**
This change will make it easier to manage our unit tests.

There is nothing special about the `test` module in rust libraries, it's just a separate module that is conditionally compiled based on whether or not the crate is running with the `test` command. As such, it's possible (and allowable) to separate the unit test code into a separate file.

It would probably be wise to move these tests closer to the code that they test, but let's do that in the future.

fixes: #82

### CODE REVIEW

**Comment by trent-reed** [comment]
Oh, forgot to mention, this is the last of my changes that I had already started on this repo; so after this I'll be moving to my fork. Just a quick FYI on that.

File: client/tests/integration_tests.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+/// command line.
+///
+/// To run locally in the docker container:
+///  `cd client && docker compose run simulator_tests`
+///
+/// These tests must be run with `--test-threads=1`, because they use a single TCP port.
+
+// Separate module to avoid us running these tests under normal `cargo test` runs.
+#[cfg(feature = "tpm-simulator-tests")]
+mod simulator_tests;
```
Comment (oddcoder) [inline client/tests/integration_tests.rs:13]: can't we just move `client/tests/simulator_tests/mod.rs` to `client/tests/simulator_tests.rs` place a 
`#![cfg(feature = "tpm-simulator-tests")]` on line 1, delete everything else in this directory and it should still work exactly the same, with less nested structure.

Comment (oddcoder) [reply client/tests/integration_tests.rs:13]: Amusing that actually works, we might want to keep the comments of this file in `client/tests/simulator_tests.rs`

**Review by oddcoder — APPROVED** [review]
everything else looks good to me.

> Oh, forgot to mention, this is the last of my changes that I had already started on this repo; so after this I'll be moving to my fork. Just a quick FYI on that.

Thanks for letting me know, I didn't actually notice until you mentioned it, my work flow is usually something along the lines:

```
wget https://github.com/tpm-rs/tpm-rs/pull/83/commits/310d60f7bc9778a703af5c5226bec987d4dbcea8.patch
```
perhaps that is why I always mix the words commit/PR/patch.

I made little suggestion, if it work, it might be nice to remove the unnecessary file. If not please ignore it. I already approved to avoid blocking you till next day.

File: client/tests/integration_tests.rs (hunk 1 above)
Comment (trent-reed) [reply client/tests/integration_tests.rs:13]: Ah yeah sorry, I didn't mention this in the PR description, but I did actually already try this and it doesn't work. :(

Comment (trent-reed) [reply client/tests/integration_tests.rs:13]: Oh wait, it might work, it might have failed because I had it after the doc comment and that is a fatal error - I didn't notice that. Let me try again.

Comment (trent-reed) [reply client/tests/integration_tests.rs]: Yup, it works! Updated the PR - I'll squash and merge. Thank you so much for the quick review!
